packages/odyssey-react/src/components/Icon/_icon-template.js

```diff
@@ -0,0 +1,79 @@
+/*!
+ * Copyright (c) 2021-present, Okta, Inc. and/or its affiliates. All rights reserved.
+ * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
+ *
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+const changeCase = require('change-case');
+
+function odysseyIconTemplate(
+  { template },
+  opts,
+  { componentName, jsx, exports },
+) {
+  const plugins = ['jsx']
+  if (opts.typescript) {
+    plugins.push('typescript')
+  }
+  const typeScriptTpl = template.smart({ plugins, preserveComments: true })
+  
+  const headerComment = `/*!
+ * Copyright (c) 2021-present, Okta, Inc. and/or its affiliates. All rights reserved.
+ * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
+ *
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+`;
+
+const title = changeCase.capitalCase(componentName.name.substring(3));
+
+  return typeScriptTpl.ast`
+${headerComment}
+import { useMemo } from 'react';
+import type { SVGProps, CSSProperties } from 'react';
+import { nanoid } from 'nanoid';
+import styles from './Icon.module.scss';
+
+interface Props {
+  title?: string;
+  titleId?: string;
+  size?: string;
+  color?: string;
+}
+
+function ${componentName}({
+  title = '${title}',
+  titleId,
+  size,
+  color,
+  ...props
+}: SVGProps<SVGSVGElement> & Props): JSX.Element {
```

Comment:
minor but can we move this type composition outside the function declaration? We can setup an extended interface or type alias above.

Also, it it possible to write these Icon components in a conventional way to our other work? For example use a `forwardRef` to the SVG element or just use a `FunctionComponent` type annotation rather than an explicit return type?